APPSERV-124 Adds sections on roles and page sync, settings image update (#6)
payara/Payara-Server-Documentation-Antora: docs/modules/ROOT/pages/documentation/user-guides/monitoring/monitoring-console.adoc

To reset all pages and remove all user created pages clear the browser's local storage for the monitoring console page.

[[monitoring-console-page-sync]]
==== Page Synchronisation
Monitoring Console uses the browser's local storage to store the page configuration (which includes the widgets placed on a page).
This means by default changes to page configurations do not affect other users.
To make pages available to others and to start from a common base configuration pages are explicitly synchronised with a persistent server page configuration.

The chosen role controls the behaviour and available options for the page synchronisation.
It is the _Administrator_'s role that is meant to manage the persistent server page configuration.
The _Administrator_ pushes local changes to the server to update the base configuration.

> _User_s and _Guest_s pull the server configuration to update their local configuration with the base.

Reviewer: I'm not sure if it's just the GitHub preview, but it doesn't like this italics syntax despite working elsewhere. Best guess is the no space between the closing `_` and the `s`

Reviewer 2: You're right, @Pandrex247, it now looks like this in Antora: ![image](https://user-images.githubusercontent.com/2195988/82217764-31fc3380-991b-11ea-86d8-e9406b3c8602.png)

@jbee, if you need to format only a part of a word, you need to double the underscores. This is called "unconstrained" formatting in the Asciidoc documentation: https://asciidoctor.org/docs/asciidoc-writers-guide/#quoted-text

This should work as expected:

```suggestion
__User__s and __Guest__s pull the server configuration to update their local configuration with the base.
```

The control remains with each user.

The _Page Synchronisation_ dialogue shown when entering the console UI allows the user to pick the pages that should be updated (pulled).

image::monitoring-console/mc_dia_pages.png[Page Synchronisation]

Based on the role the pages to synchronise are pre-selected.

* _Guest_: by default updates all pages available remotely unless they are already up to date
* _Administrator_ and _User_: by default updates all pages available remotely unless a page has been locally changed

The newest version is marked by a green background for easier orientation.

This dialogue can also be opened from the _Global_ settings by clicking the _Update Local Pages_ button shown _Page Sync_ property.

image::monitoring-console/mc_setting_global_rolesync.png[Global Page Synchronisation Settings]

> _Administrator_s also find a button to _Update Remote Pages_ with any changes applied to local changes.

Reviewer: Again, italics are going weird here. Might just be GitHub preview.

Reviewer 2: ```suggestion
__Administrator__s also find a button to _Update Remote Pages_ with any changes applied to local changes.
```

This is only needed for pages that are not using _auto_matic updates.

To share a page that so far is not part of the persistent remote configuration open the _Page_ settings and look for the _Sync_ property:

image::monitoring-console/mc_setting_page_sync.png[Page Synchronisation Settings]

_User_s and _Administrator_s can both _Push_ a page to update the persistent remote configuration and _Pull_ the remote configuration to update the local page.

> _Administrator_s can also enable or disable the _auto_matic page synchronisation.

Reviewer: See above.

Reviewer 2: ```suggestion
__User__s and __Administrator__s can both _Push_ a page to update the persistent remote configuration and _Pull_ the remote configuration to update the local page.
__Administrator__s can also enable or disable the __auto__matic page synchronisation.
```

When enabled any local change done by an _Administrator_ is automatically pushed to the remote configuration and can be pulled by other users.

[[monitoring-console-user-role]]
==== Changing User Role
The monitoring console can be used in one of three roles:

* Guest: A user that uses the console only on occasion to check existing metrics on existing pages ("read-only" user).
* User: A user that uses the console to experiment with its features without affecting others.
* Administrator:  A user that uses the console to configure the configuration that is the basis for other users.

> When no role is set for the session the user is asked to chose the role when opening monitoring console.

Reviewer: ```suggestion
When no role is set for the session the user is asked to choose a role when opening the monitoring console.
```

[[monitoring-console-settings]]
=== Global Configurations
The configuration that controls the monitoring console UI.
It is stored in the browser's local storage.

> For manual sharing the configuration can be exported and imported in the form of JSON files to share or distribute configurations.

Reviewer: ```suggestion
For manual sharing the configuration can be exported and imported in the form of JSON files.
```

> Server integrated sharing can be done using the build in page synchronisation which is a two way update mechanism to update either the local configuration with server configuration or the other way around which allows to share changes made by one user with other users in a more convenient way.

Reviewer: ```suggestion
Server integrated sharing can be done using the built in page synchronisation which is a two-way update mechanism to update either the local configuration with server configuration or the other way around which allows the sharing of changes made by one user with other users in a more convenient way.
```

==== Thread Health Page
The _Threads_ page gives detailed information on thread health of the cluster and lists information on stuck or hogging threads. Total number of live and daemon threads are given for reference.

image::monitoring-console/mc_page_threads.png[Thread Health Page]

* _Stuck Thread Incidents_: Table listing the most recent information for threads that are considered stuck. This is the case if the time passed since a thread started a not yet finished unit of work becomes longer than the set threshold in the _Stuck Threads_ health check configuration. This could be because it is waiting at a lock or the operation takes a considerable amount of time.
* _Hogging Thread Incidents_: Table listing the most recent information for threads that are considered hogging. This is the case if the thread's use of the CPU in percent points is higher than the threshold value configured in the _Hogging Threads_ health check configuration for n times in a row, where n is 1 + the configured number of retries.
* _Live Threads_: Total number of threads used by the JVM (including daemon threads)
* _Daemon Threads_: Total number of daemon threads used by the JVM

> WARNING: Be aware that the number shown in the example image are not characteristic. For demonstration purposes the thresholds have been set uncharacteristically low to be able to "force" entries. Usually the threshold for stuck threads should be in region of some seconds to a few minutes. The threshold for hogging threads should be well above 90% with a few reties.

Reviewer: ```suggestion
WARNING: Be aware that the number shown in the example image are not characteristic. For demonstration purposes the thresholds have been set uncharacteristically low to be able to "force" entries. Usually the threshold for stuck threads should be in region of some seconds to a few minutes. The threshold for hogging threads should be well above 90% with a few retries.
```
